ledger/participant-integration-api/src/main/scala/platform/store/backend/common/Field.scala

```diff
@@ -91,11 +92,21 @@ private[backend] case class BooleanOptional[FROM](extract: FROM => Option[Boolea
   override def convert: Option[Boolean] => lang.Boolean = _.map(Boolean.box).orNull
 }
 
+private[backend] object Timestamp {
```

Comment:
can it be that it would be better to just drop the timestamp, and use the bigint field instead?
* Primarily conversions need to happen in one layer above, in UpdateToDDto, because that computation is happening on a different stage, computation done here is happening on Connection holding threads. (although in this particular case it does not sound expensive)
* We anyway convert almost anythign to Instant first, and then here to bigint...would it be easier to roll an implicit decorator to daml Timestamp with .toEpocMicros and change all .toInstant to .toEpochMicros?
* slightly lower footprint in the buffers with holding longs
* smaller surface in supported fields/fieldstrategy in the Schema framework

Reply:
also for the Timestamp: it could be an implicit decorator to get out the micros from Instant-s as well for the read side

Reply:
Great suggestion. The `com.daml.lf.data.Time.Timestamp` type is just a wrapper for a Long value. We can just use `u.recordTime.micros` in UpdateToDto, no need for an implicit decorator. It saves space in DbDto values, and we don't spend time converting to Instant and back to Long.

Reply:
I have modified `DbDto`s to store `Long` values. All input timestamps in `Update` use the `com.daml.lf.data.Time.Timestamp` type, so conversion to micros is trivial.

At some point in the future, we should modify the rest of `StorageBackend` to return `com.daml.lf.data.Time.Timestamp` values instead of `Instant` values. Almost everything in Daml (except perhaps deduplication time windows?) uses microsecond resolution, converting to `Instant` and then to `com.daml.lf.data.Time.Timestamp` is just wasting resources.